### ISSUE TITLE
Handle StreamReset and ConnectionTerminated events

### ISSUE THREAD

**Issue by sethmlarson** [urllib3/urllib3#3291]
h2 has two event types for error conditions, we need to handle those and all their lower-level status codes to translate them into other errors, closing the connection, etc.

**Comment by elminson** [comment]
@sethmlarson can you share the h2 code reference ? I will love to take a look over

**Comment by sethmlarson** [comment]
@elminson Yes! It would be all the error conditions documented here: http://python-hyper.org/projects/h2/en/stable/api.html

**Comment by illia-v** [comment]
The bounty has been removed for the reasons described in https://github.com/urllib3/urllib3/issues/3000#issuecomment-5227891874.

Please refrain from opening pull requests for this issue for the time being.
